File: share-code/bai-tap-lap-trinh/python/shape/17.py
height = 3
result = ""

for i in range(1, height + 1):
    char = "=" * (height - i)
    left = ""

    for j in range(1, i + 1):
        left += str(j)

    right = left[:-1][::-1]

    result += char + left + right + char + "\n"

print(result, end="")

File: share-code/bai-tap-lap-trinh/cpp/05-thaphinh/17.cpp
#include <iostream>
#include "zvn.h"

using namespace std;

int main()
{
    /*
    ===1===
    ==121==
    =12321=
    1234321
    */
    int number = 1;
    const int HEIGHT = 5;
    string shape = "";
    for (int i = number; i <= HEIGHT; i++)
    {
        string charater = myRepeat("=", HEIGHT - i);
        string left = createListNumber(1, i - 1);
        string right = createListNumberReverse(i, 1);
        string record = charater + left + right + charater;
        shape += record + "\n";
    }
    cout << shape << "\n";
}